source/fundamentals/auth.txt

```diff
@@ -0,0 +1,384 @@
+read our :doc:`Connection Guide </fundamentals/connection>`.
+
+Specify an Authentication Mechanism
+-----------------------------------
+
+You can specify your authentication mechanism and credentials when connecting
+to MongoDB using either of the following:
+
+- a connection string
+- a ``MongoCredential`` factory method
+
+A **connection string** or **connection uri** provides instructions on how
+to connect and authenticate to your MongoDB cluster. To specify your
+authentication mechanism using a connection string, include your settings in
+your connection string and pass it to the ``MongoClients.create()`` method
+to instantiate your ``MongoClient``. For descriptions of how to use a 
+connection string to specify your authentication method, navigate to the 
+relevant mechanism section and select the :guilabel:`Connection String` tab.
+
+The ``MongoCredential`` class contains static factory methods that allow you
+to specify your authentication mechanism and credentials. You can use this
+helper class to keep your code organized. If you choose to use the 
+``MongoCredential`` helper class, you need to use the 
+``MongoClientSettings.Builder`` class to configure your connection settings 
+when constructing your ``MongoClient``. For descriptions of how to use the
+``MongoCredential`` class to specify your authentication method, navigate
+to the relevant mechanism section and select the :guilabel:`MongoCredential` 
+tab.
+
+For more information on the aforementioned classes and methods, refer to the
+following API documentation:
+
+- :java-sync-api:`MongoClients.create() </com/mongodb/client/MongoClients.html#create(com.mongodb.ConnectionString)>`
+- :java-sync-api:`MongoClient </com/mongodb/client/MongoClient.html>`
+- :java-sync-api:`MongoClientSettings.Builder </com/mongodb/MongoClientSettings.Builder.html>`
+- :java-sync-api:`MongoCredential </com/mongodb/MongoCredential.html>`
+
+Mechanisms
+----------
+
+.. _default-auth-mechanism:
+
+Default
+~~~~~~~
+
+The default authentication mechanism is a fallback setting that instructs
+the driver to negotiate the first authentication mechanism supported by the
+server in the following order of preference:
+
+#. ``SCRAM-SHA-256``
+#. ``SCRAM-SHA-1``
+#. ``MONGODB-CR``
+
+When you use the default authentication mechanism, the driver first attempts
+to authenticate using ``SCRAM-SHA-256``. If the version of the MongoDB
+instance does not support that mechanism, the driver attempts to authenticate
+using ``SCRAM-SHA-1``. If the instance does not support that mechanism either,
+the driver attempts to authenticate using ``MONGODB-CR``.
+
+The following code snippets show how to specify the connection mechanism,
+using the following placeholders:
+
+* ``username`` - your MongoDB username
+* ``password`` - your MongoDB user's password
+* ``hostname`` - network address of your MongoDB server, accessible by your client
+* ``port`` - port number of your MongoDB server
```

Comment:
You specify the default value of "admin" for ``authenticationDb`` -- should we also mention the default value of 27017 for the Mongo access port?

Reply:
Thanks for bringing this up. It's a tough call, but I think omitting instruction about the placeholders that are more related to connection than authentication is justified for keeping the description concise. I think this is a good place given there isn't any mention of the authentication db parameter in the Connection Guide (which is appropriate IMHO).

Reply:
(Good place for the authenticationDb, not so much for the default port)